xc7/primitives/slicem/Ndram/d_dram.pb_type.xml

```diff
@@ -1,48 +1,128 @@
 <pb_type name="BLK_IG-D_DRAM" num_pb="1" xmlns:xi="http://www.w3.org/2001/XInclude">
   <clock  name="CLK" num_pins="1" />
   <input  name="A"   num_pins="6" />
+  <input  name="WA"  num_pins="6" />
   <input  name="WA7" num_pins="1" />
   <input  name="WA8" num_pins="1" />
   <input  name="DI1" num_pins="1" />
   <input  name="DI2" num_pins="1" />
   <input  name="WE"  num_pins="1" />
 
+  <output name="DO6"  num_pins="1" />
+  <output name="DO6_32"  num_pins="1" />
   <output name="SO6"  num_pins="1" />
   <output name="SO6_32"  num_pins="1" />
   <output name="O6"  num_pins="1" />
   <output name="O5"  num_pins="1" />
 
- <!-- TODO: Missing modes: SRL. -->
- <mode name="64_SINGLE_PORT">
-   <xi:include href="spram64.pb_type.xml" />
-   <interconnect>
-     <direct name="CLK" input="BLK_IG-D_DRAM.CLK" output="BLK_MM-SPRAM64.CLK" />
-     <direct name="A" input="BLK_IG-D_DRAM.A" output="BLK_MM-SPRAM64.A" />
-     <direct name="WA7" input="BLK_IG-D_DRAM.WA7" output="BLK_MM-SPRAM64.WA7" />
-     <direct name="WA8" input="BLK_IG-D_DRAM.WA8" output="BLK_MM-SPRAM64.WA8" />
-     <direct name="DI" input="BLK_IG-D_DRAM.DI1" output="BLK_MM-SPRAM64.DI1" />
-     <direct name="WE" input="BLK_IG-D_DRAM.WE" output="BLK_MM-SPRAM64.WE" />
-
-     <direct name="O6" input="BLK_MM-SPRAM64.O6" output="BLK_IG-D_DRAM.O6" />
-     <direct name="SO6" input="BLK_MM-SPRAM64.O6" output="BLK_IG-D_DRAM.SO6" />
-   </interconnect>
- </mode>
- <mode name="32_SINGLE_PORT">
-   <xi:include href="spram32.pb_type.xml" />
-   <interconnect>
-     <direct name="CLK" input="BLK_IG-D_DRAM.CLK" output="BLK_MM-SPRAM32.CLK" />
-     <direct name="A" input="BLK_IG-D_DRAM.A[4:0]" output="BLK_MM-SPRAM32.A[4:0]" />
-     <direct name="DI1" input="BLK_IG-D_DRAM.DI1" output="BLK_MM-SPRAM32.DI1" />
-     <direct name="DI2" input="BLK_IG-D_DRAM.DI2" output="BLK_MM-SPRAM32.DI2" />
-     <direct name="WE" input="BLK_IG-D_DRAM.WE" output="BLK_MM-SPRAM32.WE" />
-
-     <direct name="O6" input="BLK_MM-SPRAM32.O6" output="BLK_IG-D_DRAM.O6" />
-     <direct name="O5" input="BLK_MM-SPRAM32.O5" output="BLK_IG-D_DRAM.O5" />
-     <direct name="SO6_32" input="BLK_MM-SPRAM32.O6" output="BLK_IG-D_DRAM.SO6_32" />
-   </interconnect>
- </mode>
-
- <metadata>
-  <meta name="fasm_prefix">DLUT</meta>
- </metadata>
+<!-- TODO: Missing modes: SRL. -->
+  <mode name="LUT">
```

Comment:
You can't have the LUT mode in d_dram.  When any DRAM is used in a slice, the D element must be a RAM.

Reply:
Ok, this makes sense because, as far as I have seen, the DLUT inputs `DI` are governing all the other `LUTs`. I have tried to remove it, but the routing failed as I expected. I will look again into the `clusterer` to solve the issue